libs/liblobby/lobby/lobby.lua

```diff
@@ -898,6 +898,11 @@ end
 -- Bots have isBot=true, AIs have aiLib~=nil and humans are the remaining
 -- Example: _OnUpdateUserStatus("gajop", {isAway=false, isInGame=true})
 function Lobby:_OnUpdateUserBattleStatus(userName, status)
+	if status.owner == nil and not self.users[userName] or
+	   status.owner and not self.users[status.owner] then
+		Spring.Echo("[liblobby] Error OnUpdateUserBattleStatus: Tried to update non connected user in battle: ", userName)
+		return
+	end
```

Comment:
This change (the main change of this PR?) seems unrelated and should be done in a different PR.

I also the conditions in this if are confusing. Maybe you meant to use parentheses here?

```lua
(status.owner == nil and not self.users[userName]) or
(status.owner ~= nil and not self.users[status.owner])
```